### PR TITLE
fix: changing rstudio check-idle logic

### PR DESCRIPTION
**Pull request by SanketD92**
Issue #, if available:
https://github.com/awslabs/service-workbench-on-aws/issues/185

Description of changes:
Changing check-idle logic for RStudio.

Checklist: 

* [x] Have you successfully deployed to an AWS account with your changes? 
* [x] Have you linted your code locally prior to submission?
* [x] Have you successfully ran unit tests and manual tests with your changes locally?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: main/solution/machine-images/config/infra/files/rstudio/check-idle [hunk 1]
```diff
@@ -1,42 +1,53 @@
...
+    for line in fh:
+        pass
+    last_active_time = parser.parse(line)
 
-if num_sessions or not os.path.exists(LAST_ACTIVE_TIME_FILENAME):
-    open(LAST_ACTIVE_TIME_FILENAME, 'w').write(str(current_time))
-    print('{0} RStudio sessions active at {1}'.format(num_sessions, current_time))
+# Compare with old_activity_count
+# If higher than old count, update file with time and count
+if latest_activity_count > last_activity_count:
```
Comment (Bingjiling) [inline main/solution/machine-images/config/infra/files/rstudio/check-idle:45]: Why is the condition here higher than old count? Not latest_activity_count > 0?

File: main/solution/machine-images/config/infra/files/rstudio/check-idle [hunk 2]
```diff
@@ -1,42 +1,53 @@
...
+        str(latest_activity_count) + '\n' + str(current_time))
 else:
-    last_active_time = float(open(LAST_ACTIVE_TIME_FILENAME).read())
-    idle_time = current_time - last_active_time
-    uptime = get_uptime()
-    print('RStudio has been idle for {0} seconds and system uptime is {1}'.format(idle_time, uptime))
-    if idle_time >= MAX_IDLE_SECONDS and uptime >= MAX_IDLE_SECONDS:
+    # If time difference between file time and now exceeds MAX_IDLE_MINUTES, shutdown
+    idle_minutes = (current_time - last_active_time).total_seconds()/60
+    if MAX_IDLE_MINUTES != 0 and idle_minutes >= MAX_IDLE_MINUTES:
```
Comment (Bingjiling) [inline main/solution/machine-images/config/infra/files/rstudio/check-idle:51]: Minor - Is there a way to quit the file execution early on? If MAX_IDLE_MINUTES == 0, the file don't really need to be ran.

File: main/solution/machine-images/config/infra/files/rstudio/check-idle (hunk 1 above)
Comment (SanketD92) [reply main/solution/machine-images/config/infra/files/rstudio/check-idle:45]: If there is any activity on the RStudio platform, the count would increase. Even if the instance is rebooted this count, and the log file from which the count is derived, do not change; the count and logs are simply updated and appended respectively.

File: main/solution/machine-images/config/infra/files/rstudio/check-idle (hunk 2 above)
Comment (SanketD92) [reply main/solution/machine-images/config/infra/files/rstudio/check-idle:51]: Good point. Making that change.

File: main/solution/machine-images/config/infra/files/rstudio/check-idle (hunk 1 above)
Comment (Bingjiling) [reply main/solution/machine-images/config/infra/files/rstudio/check-idle:54]: Ah! Got it. Thanks for the explanation. :)

**Review by Bingjiling — APPROVED** [review]
LGTM!

**Pull request merged**